src/families/bitcoin/wallet-btc/crypto/zec.ts

```diff
@@ -52,7 +53,13 @@ class ZCash implements ICrypto {
     account: number,
     index: number
   ): string {
-    return this.getLegacyAddress(xpub, account, index);
+    if (Base.addressCache[`${derivationMode}-${xpub}-${account}-${index}`]) {
```

Comment:
You could put `${derivationMode}-${xpub}-${account}-${index}` in a const, to avoid duplication (in other files too).